src/dodal/devices/areadetector/adaravis.py

```diff
@@ -0,0 +1,51 @@
+from nslsii.ad33 import SingleTriggerV33
+from ophyd import Component as Cpt
+from ophyd import DetectorBase
+from ophyd.areadetector.base import ADComponent as Cpt
+from ophyd.areadetector.detectors import DetectorBase
+
+from .adutils import Hdf5Writer, SyncrhonisedAdDriverBase
+
+_ACQUIRE_BUFFER_PERIOD = 0.2
+
+
+class AdAravisDetector(SingleTriggerV33, DetectorBase):
```

Comment:
TODO: Consider better ways to share functionality with `AdSimDetector` without resorting to mixins.